packages/vite/src/node/plugins/css.ts

```diff
@@ -161,7 +162,7 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
     },
 
     async transform(raw, id) {
-      if (!isCSSRequest(id) || commonjsProxyRE.test(id)) {
+      if (!isCSSRequest(id) || commonjsProxyRE.test(id) || isURLRequest(id)) {
```

Comment:
Maybe we check using [SPECIAL_QUERY_RE](https://github.com/vitejs/vite/blob/04016df8beb7110ad2087ada8e1abbefdbd70254/packages/vite/src/node/constants.ts#L22) instead?

Reply:
That would capture things other than ?url

Reply:
I think the four queries (worker, sharedworker, raw, url) captured by the regex should all be skipped for the css plugin. The former two are handled by the `workerPlugin`, the latter two are by the `assetPlugin`. Pretty sure skipping raw would be fine here too, there's not much reason to transform a raw import. It might indirectly fix #5724.

Reply:
Though if we want to scope the fix to ?url only, I'm fine with the current change too.